docs/gallery/domain/images.py

```diff
@@ -0,0 +1,270 @@
+#
+# Image data can be stored either in the HDF5 file or as an external image file.
+# For this tutorial, we will use fake image data with shape of ``('time', 'x', 'y', 'RGB') = (200, 50, 50, 3)``.
+# As in all :py:class:`~pynwb.base.TimeSeries`, the first dimension is time.
+# The second and third dimensions represent x and y.
+# The fourth dimension represents the RGB value (length of 3) for color images.
+#
+# NWB differentiates between acquired data and data that was presented as stimulus.
+# We can add it to the :py:class:`~pynwb.file.NWBFile` object as stimulus data using
+# the :py:meth:`~pynwb.file.NWBFile.add_stimulus` method.
+#
+
+image_data = np.random.randint(low=0, high=255, size=(200, 50, 50, 3), dtype=np.uint8)
+optical_series = OpticalSeries(
+    name="StimulusPresentation",  # required
+    distance=0.7,  # required
+    field_of_view=[0.2, 0.3, 0.7],  # required
+    orientation="lower left",  # required
+    data=image_data,
+    unit="n.a.",
+    format="raw",
+    starting_frame=[0.0],
+    rate=1.0,
+    comments="no comments",
+    description="The images presented to the subject as stimuli",
+)
+
+nwbfile.add_stimulus(timeseries=optical_series)
+
+####################
+# ImageSeries: Storing series of images as acquisition
+# ----------------------------------------------------
+#
+# :py:class:`~pynwb.image.ImageSeries` is a general container for time series of images acquired during
+# the experiment. Image data can be stored either in the HDF5 file or as an external image file.
+# When color images are stored in the HDF5 file the color channel order is expected to be RGB.
+#
+# We can add raw data to the :py:class:`~pynwb.file.NWBFile` object as *acquisition* using
+# the :py:meth:`~pynwb.file.NWBFile.add_acquisition` method.
+#
+
+image_data = np.random.randint(low=0, high=255, size=(200, 50, 50, 3), dtype=np.uint8)
+behavior_images = ImageSeries(
+    name="ImageSeries",
+    data=image_data,
+    description="Image data of an animal moving in environment.",
+    unit="n.a.",
+    format="raw",
+    rate=1.0,
+    starting_time=0.0,
+)
+
+nwbfile.add_acquisition(behavior_images)
+
+####################
+# External Files
+# ^^^^^^^^^^^^^^
+#
+# External files (e.g. video files of the behaving animal) can be added to the :py:class:`~pynwb.file.NWBFile` by creating
+# an :py:class:`~pynwb.image.ImageSeries` object using the ``external_file`` attribute that specifies the
+# path to the external file(s) on disk. The file(s) path must be relative to the path of the NWB file.
+# Either ``external_file`` or ``data`` must be specified, but not both.
+#
+# If the sampling rate is constant, use ``rate`` and ``strating_time`` to specify time.
+# For irregularly sampled recordings, use ``timestamps`` to specify time for each sample.
+
+external_file = [
+    os.path.relpath(movie_path, nwbfile_path) for movie_path in moviefiles_path
+]
+timestamps = [0.0, 0.04, 0.07, 0.1, 0.14, 0.16, 0.21]
```

Comment:
I have not used ImageSeries with multiple movie files before, but I think if you use explicit timestamps with movie files you probably need to specify timestamps for all frames instead of just 1 timestamp per movie file. @bendichter thoughts?

Reply:
What you have here is probably correct and I should have just looked closer at what ``moviefiles_path`` contains. Just to make it a bit clearer I would suggest adding a brief comment here (and/or where ``moviefiles_path`` is defined).